### DISCUSSION TITLE
Exporting menu in firejailed application?

### DISCUSSION THREAD

**Discussion by mike-cuche** [netblue30/firejail#4567] [Q&A] [answered]
Hello there.

I installed firejail and I'm here post into the discussion area because i think this is a not-reallly-an-issue: I'm using Plasma and an AppMenu applet which detaches the menu from the application and displays it on the panel:
![Screenshot_20210921_190259](https://user-images.githubusercontent.com/1301074/134216587-7c087fd9-1828-4d04-bbd2-124dfb847b4c.png)

However, when I run the same application with Firejail, the menu is not exported to the panel, and 
![Screenshot_20210921_190319](https://user-images.githubusercontent.com/1301074/134216788-ed348e4f-9fbb-4490-bc00-b1e6e21d044e.png)
Also, I noticed for qbittorrent that the icon in the system tray is not shown (it does show without firejail)

Is possible to fix these not-really-an-issue somehow?

**Comment by rusty-snake** [comment, marked as answer]
Both features are implemented via D-Bus. You can make it work but this also reduces security.

Simple solution: Add this to you `qbittorrent.local`:
```
ignore dbus-user none
```

More difficult but also more secure:
```
ignore dbus-user none
dbus-user filter
# Tray-icon
dbus-user.talk org.kde.StatusNotifierWatcher
dbus-user.own org.kde.*
# Global menu
dbus-user.talk _what_every_it_necessary_
```

**Reply by mike-cuche** [reply]
Yay, thanks for the tip! 
I used `dbus-user.talk org.kde.*`, it's overkill I guess, but it allowed the export of the menu bar. :)

**Reply by rusty-snake** [reply]
For my interest, does the tray-icon work if you only use `dbus-user.talk org.kde.StatusNotifierWatcher`?

**Reply by mike-cuche** [reply]
Yes, it does
